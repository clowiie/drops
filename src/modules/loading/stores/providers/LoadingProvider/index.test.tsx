import { render } from '@testing-library/react'

import mockComponent from 'test/utils/mockComponent'

import * as LoadingProviderType from '.'

describe('LoadingProvider', () => {
  const SpinnerOverlaySpy = mockComponent(
    'modules/loading/components/SpinnerOverlay',
  )

  const useLoadingSpy = jest.fn()
  jest.doMock('modules/loading/hooks/useLoading', () => useLoadingSpy)

  const Component = () => <h1>Component</h1>

  const { default: LoadingProvider } =
    require('.') as typeof LoadingProviderType

  beforeEach(() => {
    useLoadingSpy.mockReturnValue({
      loading: false,
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots with loading', () => {
      useLoadingSpy.mockReturnValue({
        loading: true,
      })

      const { container, getByText } = render(
        <LoadingProvider>
          <Component />
        </LoadingProvider>,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(SpinnerOverlaySpy).toBeCalledTimes(1)
      expect(getByText('Component')).toBeDefined()
    })

    it('should match snapshots', () => {
      const { container, getByText } = render(
        <LoadingProvider>
          <Component />
        </LoadingProvider>,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(SpinnerOverlaySpy).not.toBeCalled()
      expect(getByText('Component')).toBeDefined()
    })
  })
})
