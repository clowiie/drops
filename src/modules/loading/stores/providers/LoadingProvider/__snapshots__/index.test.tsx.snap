// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingProvider snapshots should match snapshots 1`] = `
<h1>
  Component
</h1>
`;

exports[`LoadingProvider snapshots should match snapshots with loading 1`] = `<div />`;
